### PR TITLE
Logs added to all functions, fixed.

### PR DESCRIPTION
**Pull request by yunusemrepekguc**

### CODE REVIEW

File: mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/native_advanced.kt [hunk 1]
```diff
@@ -45,32 +48,77 @@ class native_advanced : CustomEventNative() {
...
+
+        } catch (e: Exception) {
+            val stacktrace =
+                StringWriter().also { e.printStackTrace(PrintWriter(it)) }.toString().trim()
+            Log.e(
+                "TAG",
+                "Native Advanced - loadNativeAd() - Request Native Advanced Ad Failed: $stacktrace"
             )
-            return
+//            nativeAd.adListener.onAdFailed(AdParam.ErrorCode.INNER) TODO
```
Comment (oguzozsoy) [inline mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/native_advanced.kt]: Please complete TODO before PR.

File: mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt [hunk 2]
```diff
@@ -46,129 +48,184 @@ class banner : BaseAd() {
...
-        adHeight = adData.adHeight
-        val extras = adData.extras
-        mAdUnitId = extras[AD_UNIT_ID_KEY]
-
-        mHuaweiAdView = BannerView(context)
-        mHuaweiAdView.adListener = AdViewListener()
-        mHuaweiAdView.adId = mAdUnitId
-
-        val adSize: BannerAdSize? =
+        Log.d("TAG", "Banner - load()")
```
Comment (oguzozsoy) [inline mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt]: Please define the public TAG variable.

File: mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt [hunk 3]
```diff
@@ -46,129 +48,184 @@ class banner : BaseAd() {
...
+                Log.d("TAG", "Banner - load() - adData.extras => {contentUrl = $contentUrl}")
 
-        /**
-         * Prepare Child-protection keys
-         */
-        val requestConfigurationBuilder = prepareBuilderViaExtras(extras)
-        val requestConfiguration = requestConfigurationBuilder.build()
-        HwAds.setRequestOptions(requestConfiguration)
+            }else{
+                Log.e("TAG", "Banner - load() - adData.extras => contentUrl key is empty")
```
Comment (oguzozsoy) [inline mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt]: What will be done if contentUrl is empty? Does only print log enough?

File: mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt [hunk 4]
```diff
@@ -46,129 +48,184 @@ class banner : BaseAd() {
...
 
     override fun onInvalidate() {
+        Log.d("TAG", "Banner - onInvalidate()")
         Views.removeFromParent(mHuaweiAdView)
         mHuaweiAdView.adListener = null
         mHuaweiAdView.destroy()
     }
 
     override fun getLifecycleListener(): LifecycleListener? {
+        Log.d("TAG", "Banner - getLifecycleListener()")
```
Comment (oguzozsoy) [inline mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt]: Why does this method return always null?

File: mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt [hunk 5]
```diff
@@ -46,129 +48,184 @@ class banner : BaseAd() {
...
         return null
     }
 
     override fun getAdNetworkId(): String {
-        return if (mAdUnitId == null) "" else mAdUnitId!!
```
Comment (oguzozsoy) [inline mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt:116]: I think it is better to add a log for the null state

File: mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt [hunk 6]
```diff
@@ -46,129 +48,184 @@ class banner : BaseAd() {
...
     }
 
     override fun checkAndInitializeSdk(
-            launcherActivity: Activity,
-            adData: AdData
+        launcherActivity: Activity,
+        adData: AdData
     ): Boolean {
+        Log.d("TAG", "Banner - checkAndInitializeSdk()")
         return false
```
Comment (oguzozsoy) [inline mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt:172]: Could you please check the return value? It is always false.

**Review by oguzozsoy — CHANGES_REQUESTED** [review]
1. Please define the public variable for TAG.
2. Please check methods that always return the same value.

File: mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt (hunk 5 above)
Comment (yunusemrepekguc) [reply mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt:116]: There is a log inside the null check if condition.

File: mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt (hunk 3 above)
Comment (yunusemrepekguc) [reply mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt]: Related issue will be fixed in the next optimization.

File: mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt (hunk 4 above)
Comment (yunusemrepekguc) [reply mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt]: Related issue will be fixed in the next optimization.

File: mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt (hunk 6 above)
Comment (yunusemrepekguc) [reply mediation_adapter_mopub/src/main/java/com/hmscl/huawei/ads/mediation_adapter_mopub/banner.kt:172]: Related issue will be fixed in the next optimization.

**Review by umitkose1 — COMMENTED** [review]
All completed. Thanks

**Review by umitkose1 — APPROVED** [review]

**Review by umitkose1 — APPROVED** [review]

**Review by oguzozsoy — APPROVED** [review]

**Pull request merged**